main_test.go

```diff
@@ -245,6 +261,79 @@ func bincmp(ts *testscript.TestScript, neg bool, args []string) {
 	}
 }
 
+var literalGenerators = []func() *ast.BasicLit{
+	func() *ast.BasicLit {
+		buffer := make([]byte, 1+mathrand.Intn(math.MaxUint8))
+		_, err := mathrand.Read(buffer)
+		if err != nil {
+			panic(err)
+		}
+		str := base32.StdEncoding.EncodeToString(buffer)
+		return ah.StringLit(str)
+	},
+	func() *ast.BasicLit {
+		i := mathrand.Int()
+		return ah.IntLit(i)
+	},
+	func() *ast.BasicLit {
+		f := fmt.Sprint(mathrand.NormFloat64())
+		return &ast.BasicLit{
+			Kind:  token.FLOAT,
+			Value: fmt.Sprint(f),
+		}
+	},
+	func() *ast.BasicLit {
+		f := fmt.Sprint(mathrand.Float32())
+		return &ast.BasicLit{
```

Comment:
Add a FloatLit function to the asthelper package.

Don't use 2 fmt.Sprint calls.

Use https://golang.org/pkg/strconv/#FormatFloat instead of fmt.Sprint